ee/consoleuser/consoleuser_darwin.go

```diff
@@ -12,34 +12,105 @@ import (
 	"strings"
 )
 
+// example scutil output
+// ❯  scutil <<< "show State:/Users/ConsoleUser"
+// <dictionary> {
+//   GID : 20
+//   Name : jandoe
+//   SessionInfo : <array> {
+//     0 : <dictionary> {
+//       kCGSSessionAuditIDKey : 100099
+//       kCGSSessionGroupIDKey : 20
+//       kCGSSessionIDKey : 257
+//       kCGSSessionLoginwindowSafeLogin : FALSE
+//       kCGSSessionOnConsoleKey : TRUE
+//       kCGSSessionSystemSafeBoot : FALSE
+//       kCGSSessionUserIDKey : 501
+//       kCGSSessionUserNameKey : jandoe
+//       kCGSessionLoginDoneKey : TRUE
+//       kCGSessionLongUserNameKey : Janice Doe
+//       kSCSecuritySessionID : 100099
+//     }
+//     1 : <dictionary> {
+//       kCGSSessionAuditIDKey : 100999
+//       kCGSSessionGroupIDKey : 20
+//       kCGSSessionIDKey : 265
+//       kCGSSessionLoginwindowSafeLogin : FALSE
+//       kCGSSessionOnConsoleKey : FALSE
+//       kCGSSessionSystemSafeBoot : FALSE
+//       kCGSSessionUserIDKey : 502
+//       kCGSSessionUserNameKey : jrdoe
+//       kCGSessionLoginDoneKey : TRUE
+//       kCGSessionLongUserNameKey : Doe Junior
+//       kSCSecuritySessionID : 100999
+//     }
+//   }
+//   UID : 501
+// }
+
 func CurrentUids(ctx context.Context) ([]string, error) {
 	cmd := exec.CommandContext(ctx, "scutil")
 	cmd.Stdin = strings.NewReader("show State:/Users/ConsoleUser")
 
 	output, err := cmd.CombinedOutput()
 	if err != nil {
-		return nil, fmt.Errorf("executing scutl cmd: %w", err)
+		return nil, fmt.Errorf("executing scutil cmd: %w", err)
 	}
 
+	var uids []string
+
+	kCGSSessionOnConsole := ""
+	kCGSSessionUserID := ""
+
 	scanner := bufio.NewScanner(bytes.NewReader(output))
 	for scanner.Scan() {
 		line := scanner.Text()
 
-		const uidKey = "UID : "
+		// this is the name of the user at the login window
+		// kCGSSessionOnConsoleKey can still show true for other users here
+		if strings.Contains(line, "Name : loginwindow") {
+			return nil, nil
+		}
+
+		const kCGSSessionUserIDKey = "kCGSSessionUserIDKey : "
+		const kCGSSessionOnConsoleKey = "kCGSSessionOnConsoleKey : "
+
+		switch {
+		// got user id key
+		case strings.Contains(line, kCGSSessionUserIDKey):
+			kCGSSessionUserID = scutilVal(line, kCGSSessionUserIDKey)
+
+		// got session key
+		case strings.Contains(line, kCGSSessionOnConsoleKey):
+			kCGSSessionOnConsole = scutilVal(line, kCGSSessionOnConsoleKey)
 
-		if !strings.Contains(line, uidKey) {
+		default:
 			continue
 		}
 
-		parts := strings.Split(line, uidKey)
+		// if we don't have both values
+		if kCGSSessionOnConsole == "" || kCGSSessionUserID == "" {
+			continue
+		}
 
-		if len(parts) != 2 {
-			return nil, fmt.Errorf("unexpected output from scutil: %s", line)
+		// have both values
+		if kCGSSessionOnConsole == "TRUE" {
+			uids = append(uids, kCGSSessionUserID)
```

Comment:
Once we find the console user, do we need to continue scanning?

Reply:
I'm not 100% sure if there can be more than one graphical user session on macos. I've tried using the built in screen sharing app but only go control over the "desktop" rather than a user's session. However, research implies that on mac servers there can be multiple graphical user sessions.